### DISCUSSION TITLE
Could vpcloud be modified to run "in kernel space" instead of user space

### DISCUSSION THREAD

**Discussion by bmullan** [dswd/vpncloud#227] [General]
Curiosity question...

Wireguard normally runs in-kernel but there is a user space version.

The in-kernel Wireguard is incredibly faster that the user-space version.

I'm not a programmer but wondered how hard it would be (or if even possible) to build a version of
vpncloud that runs in the kernel?

**Comment by dswd** [comment]
Hi Brian,

I don't think this is feasible right now. Kernel code has a lot of restrictions, regarding e.g. memory management and threading.
Also considering the ease of installation and support for other platforms, I am not sure this is the way to go.

**Comment by bmullan** [comment]
Ok thanks
